distribution/lib/Standard/Base/0.0.0-dev/src/Enso_Cloud/Enso_File.enso

```diff
@@ -138,12 +139,123 @@ type Enso_File
+    add_label : Text -> Enso_File
+    add_label self (label : Text) -> Enso_File =
+        Context.Output.if_enabled disabled_message="Cannot add a label when writing is disabled. Press the Write button ▶ to perform the operation." panic=False <|
+            labels = self.labels
+            if labels.contains label then self else
+                ensure_tags_exist [label] . if_not_error <|
+                    new_labels = labels+[label]
+                    update_asset_labels self new_labels . if_not_error self
+
+    ## GROUP Metadata
+       ICON data_output
+       Removes a label from the file in Enso Cloud.
+
+       Arguments:
+       - label: The name of the label to remove from the file.
+
+       Returns:
+       - A boolean value indicating if the file had the label before the operation.
+
+       ? Race Conditions
+
+         Removing labels is not atomic. If two processes are modifying labels of
+         the same asset at the same time, some changes may be lost.
+    remove_label : Text -> Boolean
+    remove_label self (label : Text) -> Boolean =
+        Context.Output.if_enabled disabled_message="Cannot remove a label when writing is disabled. Press the Write button ▶ to perform the operation." panic=False <|
+            labels = self.labels
+            if labels.contains label . not then False else
+                new_labels = labels.filter l-> l != label
+                update_asset_labels self new_labels . if_not_error True
+
+    ## GROUP Metadata
+       ICON data_output
+       Sets the set of labels associated with the file in Enso Cloud.
+       Any other labels than the ones provided are removed from the file.
+
+       If any of the provided labels did not exist before, it is created with
+       a random color associated with it. If you want to set a specific color,
+       use `Enso_File.create_label` before calling this method.
+
+       Arguments:
+       - labels: The new set of labels to associate with the file.
+
+       Returns:
+       - The file on which it was called.
+    set_labels : Vector Text -> Enso_File
+    set_labels self (labels : Vector Text) -> Enso_File =
+        Context.Output.if_enabled disabled_message="Cannot update labels when writing is disabled. Press the Write button ▶ to perform the operation." panic=False <|
+            ensure_tags_exist labels . if_not_error <|
+                update_asset_labels self labels . if_not_error self
+
+    ## GROUP Metadata
+       ICON data_output
+       Creates a new label with the given name and color.
+
+       Arguments:
+       - name: The name of the label to create.
+       - color: The color of the label to create.
+    create_label : Text -> Text -> Nothing
+    create_label (name : Text) (color : Color) =
+        Context.Output.if_enabled disabled_message="Cannot create a label when writing is disabled. Press the Write button ▶ to perform the operation." panic=False <|
+            create_tag name color . if_not_error Nothing
```

Comment:
worth testing existance?

Reply:
OK. Although ideally that should be checked on the cloud side and only a special error returned.